astarte/device/pairing_handler.py

```diff
@@ -65,6 +65,32 @@ def register_device_with_jwt_token(device_id, realm, jwt_token,
         pairing_base_url)
 
 
+def generate_device_id(namespace: UUID = None, unique_data: str = None):
```

Comment:
I'm not sure if having a single function here is the best way vs having a `generate_random_device_id` and `generate_deterministic_device_id` or something like that.
As it is right now, imho it's too easy to forget a parameter and get back what you think it's a deterministic device ID (since you passed some data inside) while it's in fact a random device ID

Reply:
We do not provide a defaulting behavior for the generation of device IDs (see [the SDKs doc page](https://docs.astarte-platform.org/latest/041-astarte_device_sdks.html#device-id)), even in languages supporting default parameters. Some SDKs, e.g. Java, use overloaded methods but the preferred pattern is to have separate functions for random and UUIDv5 IDs.
I prefer the second choice, however I'm not  an expert on what is the most _pythonic_ way.